wpiutil/src/main/native/cpp/json_serializer.h

```diff
@@ -121,7 +121,7 @@ class json::serializer
             return;
         }
 
-        const bool is_negative = (x <= 0) and (x != 0);  // see issue #755
+        const bool is_negative = x < 0;  // see issue #755
```

Comment:
See https://github.com/nlohmann/json/issues/755#issuecomment-440739799

Reply:
The problem is NumberType is templated and we don't know if it's a signed or an unsigned number here and need to do the right thing regardless of if it's signed or unsigned.